io/pileup/pileup.go

```diff
@@ -49,11 +48,6 @@ import (
 
 // https://en.wikipedia.org/wiki/Pileup_format
 
-var (
-	gzipReaderFn = gzip.NewReader
```

Comment:
I think we started doing this to make it easier to mock tests?

Reply:
I think the big changes to the parsers remove this anyway - since gzip gets tossed into generics, its easier to test

Reply:
> I think we started doing this to make it easier to mock tests?

IMO, you shouldn't need to reach into a module and change its state to mock it, especially given how Go handles errors. We should favor onion architecture keep I/O as high up in the call chain as possible. This allows inner functions to be pure. The `gzipReaderFn` variable was unused anyway, that's why I removed it.

Reply:
Go's interfaces are the solution for testing, IMO. If a function/object needs to perform I/O operations, it should take `io.Reader` and `io.Writer` interfaces as an argument/field. These interfaces can then be easily be mocked:

```
import "bufio"
import "bytes"

const someData = ";alskjdf;lkajsdlf"

func main() {
    var b bytes.Buffer
    mockedWriter := bufio.NewWriter(&b)
    FunctionThatTakesAWriter(mockedWriter)
    got := b.String()

    mockedReader := bytes.NewReader([]byte(someData))
    FunctionThatTakesAReader(mockedReader)
}
```

Reply:
I agree here! Much more readable too. This is how the `bio` parsers work, and is how I mock test to completion `fasta` in ioToBio.